pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -228,12 +234,20 @@ func buildInboundHTTPFilter(mesh *meshconfig.MeshConfig, node *model.Proxy, attr
 				},
 			},
 			MixerAttributes: &mpb.Attributes{Attributes: attrs},
-			Transport:       buildTransport(mesh, attrUID(node)),
+			Transport:       buildTransport(mesh, node, attrUID(node)),
 		}),
 	}
 }
 
+func has10Config(node *model.Proxy) bool {
+	_, found := node.Metadata["ISTIO_PROXY_VERSION"]
+	return found
+}
+
 func modifyOutboundRouteConfig(in *plugin.InputParams, httpRoute route.Route) route.Route {
+	if !has10Config(in.Node) {
+		return httpRoute
+	}
```

Comment:
why this short circuiting? we had per-filter config stuff in 0.8

Reply:
@qiwzhang / @kyessenov ^^^ .. I distinctly remember that the mixer was the first to get the per-route filter config.